vllm/utils.py

```diff
@@ -42,3 +43,49 @@ def random_uuid() -> str:
 def in_wsl() -> bool:
     # Reference: https://github.com/microsoft/WSL/issues/4071
     return "microsoft" in " ".join(uname()).lower()
+
+
+def get_max_shared_mem_bytes() -> int:
+    """Gets the max optin shared memory per block for the current GPU."""
+    libnames = ("libcuda.so", "libcuda.dylib", "nvcuda.dll", "cuda.dll")
+    for libname in libnames:
+        try:
+            cuda = ctypes.CDLL(libname)
+        except OSError:
+            continue
+        else:
+            break
+    else:
+        raise OSError(f"Could not load any of {' '.join(libnames)}")
+
+    smem_size = ctypes.c_size_t()
+    device = ctypes.c_size_t()
+
+    cuda.cuDeviceGet(ctypes.byref(device), torch.cuda.current_device())
+    cuda.cuInit(0)
+    # https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__TYPES.html
+    cudaDevAttrMaxSharedMemoryPerBlockOptin = 97  # pylint: disable=invalid-name
+    assert not cuda.cuDeviceGetAttribute(
+        ctypes.byref(smem_size), cudaDevAttrMaxSharedMemoryPerBlockOptin,
+        device)
+    return smem_size.value
+
+
+def check_if_can_support_max_seq_len(max_seq_len: int,
+                                     block_size: int) -> None:
```

Comment:
If this will be only executed by `Worker`, is there reason to put this under `utils`? What about moving this function to `worker.py`?